kraken-build/src/kraken/std/cargo/tasks/cargo_publish_task.py

```diff
@@ -37,6 +42,35 @@ class CargoPublishTask(CargoBuildTask):
     #: Cargo.toml which to temporarily bump
     cargo_toml_file: Property[Path] = Property.default("Config.toml")
 
+    #: Allow Overwrite of existing packages
+    allow_overwrite: Property[bool] = Property.default(False)
+
+    def prepare(self) -> TaskStatus | None:
+        """Checks if the crate@version already exists in the registry. If so, the task will be skipped"""
+        if self.allow_overwrite.get():
+            return TaskStatus.pending()
+
+        manifest = CargoManifest.read(self.cargo_toml_file.get())
+        manifest_package = manifest.package
+        manifest_package_name = manifest_package.name if manifest_package is not None else None
+        manifest_version = manifest_package.version if manifest_package is not None else None
+
+        package_name = self.package_name.get() or manifest_package_name
+        version = self.version.get() or manifest_version
+
+        if not package_name:
+            return TaskStatus.pending("Unable to verify package existence - unknown package name")
+        if not version:
+            return TaskStatus.pending("Unable to verify package existence - unknown version")
+
+        try:
+            return self._check_package_existence(package_name, version, self.registry.get())
+        except Exception as e:
+            logger.warn(
+                f"An error happened while checking for {package_name} existence in {self.registry.get().alias}, {e}",
```

Comment:
```suggestion
                "An error happened while checking for {} existence in %s, %s",
                package_name,
                self.registry.get().alias,
                e,
```

Formatting the log message directly is discouraged as it evaluates eagerly and can lead to substitution errors if the resulting message string contains placeholders.